lib/proxy.js

```diff
@@ -272,6 +282,16 @@ ReverseProxy.prototype.setupHttpsProxy = function (proxy, websocketsUpgrade, log
     var src = _this._getSource(req);
     var httpProxyOpts = Object.assign({}, _this.opts.httpProxy);
 
+    if (opts.whitelistedIps) {
```

Comment:
You should refactor this code sin it is the same as the one starting in line 182.